compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AMDAIETargetCDODirect.cpp

```diff
@@ -198,6 +198,39 @@ LogicalResult configureBdInBlock(XAie_DmaDesc &dmaTileBd, Block &block,
                             lenInBytes);
   }
 
+  // ND zero padding.
+  std::optional<llvm::ArrayRef<BDPadLayoutAttr>> padDims =
+      bdOp.getPadDimensions();
+  if (padDims) {
+    XAie_DmaPadTensor dmaPadTensor = {};
+    dmaPadTensor.NumDim = padDims->size();
+    dmaPadTensor.PadDesc = static_cast<XAie_PadDesc *>(
+        calloc(dmaPadTensor.NumDim, sizeof(XAie_PadDesc)));
+    if (!dmaPadTensor.PadDesc)
+      return bdOp.emitError("couldn't allocate array of XAie_PadDesc");
+    // libxaie requires stride in multiples of 32b
+    double elementWidthIn32bWords =
+        static_cast<double>(bdOp.getBufferElementTypeWidthInBytes()) / 4.0;
+    for (size_t i = 0; i < padDims->size(); i++) {
+      // Pass down dimensions in reverse order.
+      int j = padDims->size() - i - 1;
+      uint8_t before;
+      uint8_t after;
+      if (j > 0) {
+        before = static_cast<uint8_t>(padDims.value()[i].getConstPadBefore());
+        after = static_cast<uint8_t>(padDims.value()[i].getConstPadAfter());
+      } else {
+        before = static_cast<uint8_t>(padDims.value()[i].getConstPadBefore() *
```

Comment:
Shouldn't we check that `(padDims.value()[i].getConstPadBefore() * elementWidthIn32bWords)  % 4 == 0` and fail otherwise? Otherwise, this could lead to hard to debug incorrect results.

Reply:
Edit: As discussed offline `(padDims.value()[i].getConstPadBefore() * elementWidthIn32bWords) % 4 == 0` is incorrect. It should be `bdOp.getBufferElementTypeWidthInBytes() * padDims.value()[i].getConstPadBefore() % 4 == 0`